### ISSUE TITLE
Allow documentation to be read from an external file

### ISSUE THREAD

**Issue by huonw** [rust-lang/rust#15470]
I.e. the following are equivalent:

lib.rs

``` rust
//! Foo bar baz
```

---

lib.rs

``` rust
#![doc(file = "extensive_crate_docs.md")]
```

extensive_crate_docs.md:

``` md
Foo bar baz
```

This could be used in the rust source for modules like [`borrowck::doc`](https://github.com/rust-lang/rust/blob/master/src/librustc/middle/borrowck/doc.rs): having that as a separate `.../borrowck/doc.md` file with `#[doc(file="doc.md")] mod doc {}` in `.../borrowck/mod.rs` might be nice (it would certainly make editing it more natural, as editors will automatically select the Markdown mode).

**Comment by steveklabnik** [comment]
Triage: https://github.com/rust-lang/cargo/issues/739 is related, but not the same.

**Comment by euclio** [comment]
What's the status on this? Does it need an RFC or are we waiting on an in-tree implementation?

**Comment by QuietMisdreavus** [comment]
This is an RFC now: https://github.com/rust-lang/rfcs/pull/1990

Discussion on it is fairly positive, so it's likely that something to this effect is going to happen.

**Comment by chriskrycho** [comment]
For anyone else who lands here via search: [the RFC was merged](https://github.com/rust-lang/rfcs/pull/1990); tracking issue is [here](https://github.com/rust-lang/rust/issues/44732).

**Comment by estebank** [comment]
Closing in favor of tracking issue #44732.

**Comment by jonas-schievink** [comment]
@estebank you didn't actually close this :P
